サーバー側のクチコミ・画像保存、リレーションの追加

Ako-in opened the pull request:
StoreControllerの修正
ImagestoreControllerの追加（画像保存用）
Routeの追加

Ako-in committed:
サーバー側のクチコミ・画像保存、リレーションの追加

StoreControllerの修正
ImagestoreControllerの追加（画像保存用）
Routeの追加

chihiro0415 commented on app/Store.php:
```diff
@@ -6,5 +6,9 @@
+    function reviews()
+    {
+        // リレーション StoreモデルにReviewは属する
+        return $this -> belongsTo('App\Review');
```
Store : Review = 1 : 多
ですよね？ 👀 

１つのお店が複数のレビュー（口コミ）をもっているはずなので
ここは
```suggestion
        return $this -> hasMany('App\Review');
```
じゃないですかね？
（Review modelの方も逆）

Ako-in replied:
修正しましたが、すでに中川さんが上げてマージ済みですので削除します。

chihiro0415 commented on app/Review.php:
```diff
@@ -6,6 +6,10 @@
+    function stores()
+    {
+        // リレーション StoreモデルはReviewの親
+        return $this -> hasMany('App\Store');
```
Store model同様

Ako-in replied:
修正しましたが、すでに中川さんが上げてマージ済みですので削除します。

chihiro0415 commented on routes/web.php:
```diff
@@ -1,5 +1,7 @@
 <?php
 
+use App\Http\Controllers\StoreController;
+use App\Http\Controllers\ReviewController;
```
👀 ？？
route にはいらないですー！

chihiro0415 commented on routes/web.php:
```diff
@@ -20,3 +22,9 @@
 
 Route::get('/home', 'HomeController@index')->name('home');
+
+Route::post('/sweets','StoreController@store')->name('sweets.store');
+Route::post('/sweets/{sweet}','ReviewController@show')->name('sweets.show');
+
+//Image保存
+Route::post('/sweets', 'ImagestoreController@store')->name('sweets.store');
```
リクエストpathが違いそうです！

Ako-in committed:
Routeのパス修正、リレーション修正

画像保存は一旦外しました。

Ako-in committed:
不要ファイルの削除　ImagestoreController

Ako-in committed:
ReviewControllerにcreateメソッドの追加、リレーションStoreメソッドの修正

リレーション：Function stores->store(単数系へ)

chihiro0415 commented on app/Http/Controllers/StoreController.php:
```diff
@@ -34,7 +36,26 @@ public function create()
     public function store(Request $request)
     {
-        //
+        $store = New Store;
+        $store -> name = $request -> name;
+        $store -> category_id = $request -> category_id;
+        $store -> prefecture_id = $request -> prefecture_id;
+        $store -> address = $request -> address;
+        $store -> phone_number = $request -> phone_number;
+        $store -> opening_hour = $request -> opening_hour;
+        $store -> closing_day = $request -> closing_day;
+        $store -> price = $request -> price;
+        $store -> store_image = $request -> store_image;
+        $store -> recommend_image = $request -> recommend_image;
+        $store -> recommend_product = $request -> recommend_product;
+        $store -> latitude = $request -> latitude;
+        $store -> longitude = $request -> longitude;
+        $store -> url = $request -> url;
+        $store -> user_id = Auth::id();
+
+        $store -> save;
+        return redirect()->route('/sweets/stores.show');
+
```
お店情報の保存処理は今回必要なさそうですが、この部分は+αでつくったみたいな感じでしょうか...？:eyes:

Ako-in replied:
お店情報は別データから持ってくるんでしたね。忘れてました。。。
今回は57行目だけ残せばいいですか？
Return view（’/sweets/stores.show);だけですか？

chihiro0415 commented on routes/web.php:
```diff
@@ -20,3 +21,11 @@
 
 Route::get('/home', 'HomeController@index')->name('home');
+
+Route::get('/sweets','StoreController@store')->name('sweets.store');
```
```suggestion
Route::post('/','StoreController@store')->name('stores.store');
```

Ako-in committed:
口コミ詳細画面、

Controller のRoute Nameの修正
ReviewController@storeの作成
StoreController@storeの削除（お店情報の入力画面がないので不要）
リレーションの削除（既にマスターにある為、Conflictが起こらないように削除）

chihiro0415 commented on app/Http/Controllers/ReviewController.php:
```diff
@@ -25,7 +25,7 @@ public function index()
     public function create()
     {
-        //
+        return view('reviews.create');
```
createページに必要な情報を返してあげてほしいです...！
今回の場合だとお店の情報ですね！
どのお店に口コミ投稿するか、ってこの時点でわかっていることなので
viewからお店のIDをもらう想定にしておいて、引数で受け取ったIDからお店のデータをfindしてきてください！

Ako-in replied:
修正したつもりですが、これであってますでしょうか？

chihiro0415 commented on app/Http/Controllers/ReviewController.php:
```diff
@@ -36,7 +36,27 @@ public function create()
+        $review -> id;
+        $review -> store_id = $request -> store_id;
+        $review -> comment = $request -> comment;
+        $review -> created_at = $request -> created_at;
```
blogアプリでやった時のようにcreated_atは自動で入ったはずなので、ここはいらないです！

chihiro0415 replied:
ここ変更されてないですね？！
（別ファイルで修正したのかな 🤔  ）

chihiro0415 commented on app/Http/Controllers/ReviewController.php:
```diff
@@ -36,7 +36,27 @@ public function create()
+        $review -> category_id = $request -> category_id;
+        $review -> post_image = $request -> post_image;
+        $review -> published_at = $request -> published_at;
+        $review -> updated_at = $request -> updated_at;
```
こちらもcreated_atと同じくです！

chihiro0415 commented on app/Http/Controllers/ReviewController.php:
```diff
@@ -36,7 +36,27 @@ public function create()
+        $review -> ate_thing = $request -> ate_thing;
+        $review -> category_id = $request -> category_id;
+        $review -> post_image = $request -> post_image;
+        $review -> published_at = $request -> published_at;
```
下書きか公開か、はviewから何かフラグをもらわないとですね！:eyes:
そのフラグによって時間を入れるか入れないか、にifで切り分けましょう！

あと `published_at ` はviewから送られてくるのではなくここで時間生成しましょうか！
```suggestion
        $review -> published_at = now();
```

Ako-in replied:
すみません、これは未対応です。

chihiro0415 commented on app/Http/Controllers/ReviewController.php:
```diff
@@ -36,7 +36,27 @@ public function create()
-        //
+        $review = new Review;
+
+        $review -> id;
```
？？

Ako-in replied:
余計なものが入っていたので削除しました。

chihiro0415 commented on app/Http/Controllers/ReviewController.php:
```diff
@@ -36,7 +36,27 @@ public function create()
+
+        $review -> save();
+        
+        $review = Review::All($id);
```
？？？

Ako-in replied:
Review::findのつもりで書いていたのですが、間違っているので修正しました。

chihiro0415 commented on app/Review.php:
```diff
@@ -7,5 +7,5 @@
 {
     //
-    
 }
+　
```
変更箇所がないファイルはコミットされないように気をつけてください〜〜！

Ako-in replied:
承知しました。

ArisaNAKAMURA-GitHub commented on app/Http/Controllers/ReviewController.php:
```diff
@@ -36,7 +36,27 @@ public function create()
     public function store(Request $request)
     {
-        //
+        $review = new Review;
```
［MUST］ここでインスタンス化されていますが、VScodeで赤色の下波線が出ています。このファイルの上の方でAuthを使えるようにしているみたいに、クラス（モデル）をuse\App\～で使えるようにしないとインスタンス化できないと思われます。ここだったらStoreとReviewが必要かなー？と考えました。

Ako-in replied:
コメントありがとうございます。修正加えました。

Ako-in committed:
ReviewController@Createの修正、 Review.phpの削除、Conflictになっている部分の解消

ReviewController@Createにお店情報を返す
Review.php 空ファイルが上がっていたので削除
Controller内のconflict部分の修正

// 下書きか公開か、はviewから何かフラグをもらわないとですね！👀
// そのフラグによって時間を入れるか入れないか、にifで切り分けましょう！
は未対応です。

chihiro0415 commented on app/Http/Controllers/ReviewController.php:
```diff
@@ -23,9 +25,27 @@ public function index()
-        //
+        $store = Store::find($id);
+        
+        $store -> store_id = $request -> store_id;
+        $store -> name = $request -> name;
+        $store -> category_id = $request -> category_id;
+        $store -> address = $request -> address;
+        $store -> opening_hour = $request -> opening_hour;
+        $store -> closing_day = $request -> closing_day;
+        $store -> phone_number = $request -> phone_number;
+        $store -> url = $request -> url;
+        $store -> price = $request -> price;
+        $store -> category_id = $request -> category_id;
+        $store -> prefecture_id = $request -> prefecture_id;
+        $store -> store_image = $request -> store_image;
+        $store -> latitude = $request -> latitude;
+        $store -> longitude = $request -> longitude;
+        $store -> save();
```
データのとってきかた、授業でやった内容はこんな感じじゃなかたはずです〜〜〜〜〜！ 😂

chihiro0415 replied:
Store@show ができているのになぜでしょう...？！？？

chihiro0415 commented on app/Http/Controllers/ReviewController.php:
```diff
@@ -36,7 +56,26 @@ public function create()
+
+        $review -> save();
+        
+        $review = Review::find($id);
```
この1行はなんでしょう...？:eyes:

Ako-in committed:
下書き保存機能を修正しました。

chihiro0415 commented on app/Http/Controllers/ReviewController.php:
```diff
@@ -36,7 +56,26 @@ public function create()
+        $review -> post_image = $request -> post_image;
+        
+        if($request->is_published){
+            $review -> published_at = now()
```
```suggestion
            $review -> published_at = now();
```

tai-namechan commented:
コンフリクトが起きてるので解決お願いします
ReviewControllerの
use App\Review; とuse App\Store;はmasterにもうあるので消して大丈夫です

Ako-in commented:
了解です！

tai-namechan commented:
すみません。。。pr確認用に切ったブランチ消そうと思ったら井上さんのprがclosedされちゃいました🙇‍♂️
reopenします。

Ako-in committed:
セミコロン記載漏れ訂正

Ako-in committed:
masterからPullしてきたファイルのコミット

chihiro0415 committed:
Merge branch 'master' into Show_review_server

chihiro0415 commented on app/Http/Controllers/ReviewController.php:
```diff
@@ -26,9 +28,27 @@ public function index()
-        //
+        $store = Store::find($id);
+        
+        $store -> store_id = $request -> store_id;
+        $store -> name = $request -> name;
+        $store -> category_id = $request -> category_id;
+        $store -> address = $request -> address;
+        $store -> opening_hour = $request -> opening_hour;
+        $store -> closing_day = $request -> closing_day;
+        $store -> phone_number = $request -> phone_number;
+        $store -> url = $request -> url;
+        $store -> price = $request -> price;
+        $store -> category_id = $request -> category_id;
+        $store -> prefecture_id = $request -> prefecture_id;
+        $store -> store_image = $request -> store_image;
+        $store -> latitude = $request -> latitude;
+        $store -> longitude = $request -> longitude;
+        $store -> save();
```
この部分削除で！

chihiro0415 commented on app/Http/Controllers/ReviewController.php:
```diff
@@ -26,9 +28,27 @@ public function index()
+        $store -> longitude = $request -> longitude;
+        $store -> save();
+
+        return view('reviews.create');
```
```suggestion
       return view('reviews.create',['store'=>$store]);
```

Ako-in committed:
不要な記述の削除

ReviewController、StoreController内で
Use App\Review;
Use App\Store;
ダブルで入力されていたため削除。

ReviewController
お店情報は引っ張ってくる必要がないため削除。

chihiro0415 reviewed (approved):
おけです！！

Ako-in commented:
マージしました！ありがとうございました！
